### DISCUSSION TITLE
Usability

### DISCUSSION THREAD

**Discussion by patagoniahiker** [Dotnet-Boxed/Templates#77] [Q&A]
Hi, in our class we are using your boilerplate in this semester, we updated to the latest update and wanted to share some information that might better help adoption/us. 

While your site's help is way better that some of the authors... 

Are you planning on including ASP MVC Site Map Provider, its a good plugin
- [ ] It would help if we had a list of variable to rename or a view to rename things like project name, site title etc
- [ ] How can we turn off some features while leaving others on, for e.g. we don't need the trailing slash, or the routing (we see this on the help page with screen view, but where is this, and is it stored in DB or another config file)
- [ ] Is the update a straight update via nuget or will this brake any of the other changes
- [ ] What routes will your routing match - this is a turning out to be a little grief for us to follow
- [ ] Where is the IIS configuration on DOS setup
- [ ] With NWEBSEC, do we need the additional filters if we setup inside the web.config

thanks for this wonderful plugin
patty

**Comment by RehanSaeed** [comment]
Wow, you are using it in your class. Is that at a university, college?

I looked at that sitemap provider. Not sure if it supports ASP.NET Core MVC 6 (that is the project template I am focusing on now). I am aware of it and found it to be overly complex at the time I was thinking about whether or not to integrate it.

Well, project name is whatever you name the project when you create it but having a configurable site name is a good idea.

I don't think you have tried the ASP.NET Core MVC 6 template, it gives you lots of options to turn things on and off. The trailing slash on the end is a search engine optimization (SEO) feature and you can either turn it on or off as it's personal preference. There are comments in place for that.

Unfortunately, when I add features to the project template, your project that you created does not get updated. However, when I update the boilerplate NuGet packages, it's easy for you to update them in your project.

Not sure what you mean by route matching. There are good tutorials on attribute routing at asp.net.

Not sure what you mean by IIS configuration on DOS.

You can use NWebSec in a number of ways, including using the Web.config file but I would not recommend it. It is better to use the filters as I have done or use the middleware.

Hope that helps!

**Comment by patagoniahiker** [comment]
yes, I guess you could say a different way of learning, using it in small learning class..  :) not the big univs. given the comments and security - this is more apt for the students!

we are still using 4.6 since 5.0/6.0 is still beta and very confusing to follow.

We have been reading your site on the security, and one suggestion is that while reading from a developers point of view who is just picking up your template, reading flow-jumps around some. So, its just a little hard to follow. I like your HTML checklist page, something organized like that for security alone would be a good start. For e.g. you mentioned IIS and DOS in your template, but it does not show anywhere in the documentation.

It would be nice to see a sample of Googles re-captcha as well since that security, its on github code developed as a plugin.

**Comment by RehanSaeed** [comment]
Ah, you want to look at the dynamicIpSecurity section in the Web.config file. It talks about DoS attacks and has a link. You may also want to use something like Cloudflare which is a lot more powerful.

In what context would I add a captcha? The basic template does not have anything to protect.

**Comment by patagoniahiker** [comment]
ok with the cpatcha... I can see your point. We are adding the ASP Identity system... with login (which comes with the default VS ASP mvc template).. and  we are replacing that with your template... so, I will need to figure that out.

would it be an _option _in the webconfig file and filter option on action... or some way to build it into a view at the routes handling section

**Comment by RehanSaeed** [comment]
Do you mean how to add a captcha or ASP.NET Identity?

**Comment by patagoniahiker** [comment]
yes, making recaptcha available as an option. Its inline with all the other security options, using your plugin to add recaptcha to forms like the modal forms .. [here is link ](http://bit.ly/1PMi4Xu) its a working link

Wanted to let you know, your nuget install is older than your template, inside nuget it says 1.0.23 from Jan this yr, but on the github page it says 2.xx..x on the images folder... not sure what to make of it.

You've SRI HASH generation for ASP MVC 6, but not the current version for 4.6, how can we invoke that helper to gen up the hash

**Comment by RehanSaeed** [comment]
This project does not contain any forms. If I were to add a captcha, I would have to add an example of a form too. and I have no plans on doing that at the moment, it's not a feature I'm getting any requests for but I'll keep it in mind.

Yes, the NuGet package has not changed for MVC 5 in a couple of months because it did not need any updates.

At some point I will probably back port the HTML helper to MVC 5 but I don't have time right now. I would accept a pull request for this work.

**Comment by patagoniahiker** [comment]
dont know how to do it, we are not experts otherwise we would be doing this ourselves.. :+1:

**Comment by RehanSaeed** [comment]
I'll add your idea to the TODO list.
